src/wrapper/jnienv.rs

```diff
@@ -2713,23 +2951,26 @@ impl<'local> JNIEnv<'local> {
         })
     }
 
-    /// Returns underlying `sys::JNIEnv` interface.
-    pub fn get_native_interface(&self) -> *mut sys::JNIEnv {
-        self.internal
-    }
-
     /// Returns the Java VM interface.
     pub fn get_java_vm(&self) -> Result<JavaVM> {
         let mut raw = ptr::null_mut();
-        let res = jni_unchecked!(self.internal, GetJavaVM, &mut raw);
+        let res = unsafe { jni_call_unchecked!(self, v1_1, GetJavaVM, &mut raw) };
         jni_error_code_to_result(res)?;
         unsafe { JavaVM::from_raw(raw) }
     }
 
     /// Ensures that at least a given number of local references can be created
     /// in the current thread.
-    pub fn ensure_local_capacity(&self, capacity: jint) -> Result<()> {
-        jni_void_call!(self.internal, EnsureLocalCapacity, capacity);
+    pub fn ensure_local_capacity(&self, capacity: usize) -> Result<()> {
```

Comment:
Using a `jint` would be a more literal/raw binding of the underlying C JNI API but it implicitly doesn't really make sense to be able to ask for a negative capacity so it seems reasonable to deviate from the C API here for ergonomics.

I don't think it would generally be realistic to even expect that values near to `i32::MAX` would be likely to succeed, the choice to use `usize` here is really just because it seems like the appropriate the Rusty type for specifying this kind of size.

Given the choice between `u32` or `usize` then we'd still have to account for a value potentially > `i32::MAX` either way but `usize` feels more appropriate here.

If we have cases like negative values that could return an error without necessarily throwing an OutOfMemory exception then it seems like anyway the caller can't really just assume that an OutOfMemory exception is the only failure mode here.

It probably would have been better to change separately though.

Sort of inclined to leave this atm, unless you feel strongly about this?

Reply:
I took another look at this, and have changed my mind.

[HotSpot also returns an error if a bogus maximum capacity is given](https://github.com/openjdk/jdk/blob/09dad0e96b37e3fcd1a13040e0de85ebc04b07c2/src/hotspot/share/prims/jni.cpp#L725), and [the default upper limit is well below `i32::MAX`](https://github.com/openjdk/jdk/blob/09dad0e96b37e3fcd1a13040e0de85ebc04b07c2/src/hotspot/share/runtime/globals.hpp#L781), so this proposed behavior is consistent with HotSpot.

It probably should be noted in this method's documentation that an excessively large capacity will result in an error, and that HotSpot's default maximum capacity is 65536.